### PR TITLE
[Mono] Enable emitting intrinsics for System.Numerics.Vector<T> on arm64

### PR DESCRIPTION
**Pull request by fanyang-mono**
Contribute to #64072

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -3501,6 +3501,10 @@ mono_emit_simd_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 	if (!strcmp (class_ns, "System.Numerics") && !strcmp (class_name, "Vector")){
 		return emit_sri_vector (cfg, cmethod, fsig, args);
 	}
+
+	if (!strcmp (class_ns, "System.Numerics") && !strcmp (class_name, "Vector`1")){
+		return emit_vector64_vector128_t (cfg, cmethod, fsig, args);
+	}
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:3507]: Aren't these types called Vector64<T> and Vector128<T> ?

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:3507]: No, they don't. This is what I observed, without this change:
This is my app https://gist.github.com/fanyang-mono/032126329c3052e8cc2d074af3cf4469
This is the codegen log for System.Numerics.Vector`1<int16>:op_Addition https://gist.github.com/fanyang-mono/4704bdc2b426aae4a2594afddb43e02c
This is the codegen log for System.Runtime.Intrinsics.Vector128`1<int16>:op_Addition https://gist.github.com/fanyang-mono/a9117311d829ba5433f6a11baf9befcb

You can see the add operator of Vector128 got replaced with intrinsics, but not the Vector one.

Comment (vargaz) [reply src/mono/mono/mini/simd-intrinsics.c:3507]: It calls emit_vector64_vector128_t (), are the Vector<T> intrinsics the same as the Vector64<T> etc ones ?

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:3507]: On arm64, vector is equivalent to vector128. On amd64, vector is either equivalent to vector128 or vector256.

**Review by vargaz — APPROVED** [review]

**Comment by fanyang-mono** [comment]
CI failures are irrelevant to this PR.

**Pull request merged**